orangecontrib/prototypes/widgets/owgrep.py

```diff
@@ -0,0 +1,151 @@
+        if int(ev.key()) == int(Qt.Key_C) and int(ev.modifiers()) == int(Qt.ControlModifier):
+#        if ev.matches(QKeySequence.Copy):
+            print("foo")
+            self.copy()
+        super().keyPressEvent(ev)
+
+class OWGrep(OWWidget):
+    name = "Grep"
+    description = "foo"
+    icon = "icons/Grep.svg"
+    priority = 1102
+
+    class Outputs:
+        data = Output("Data", Table, default=True)
+
+    text = Setting("")
+    skip_lines = Setting(0)
+    use_lines = Setting(0)
+    column_names = Setting(True)
+    auto_send = Setting(True)
+    find_text = Setting("")
+
+    re_number = re.compile(r"^([+-]?\d+([.,]\d+)?)$")
+
+    class Warning(OWWidget.Warning):
+        no_lines = Msg("Pattern not found")
+
+    class Error(OWWidget.Error):
+        different_length = Msg("Lines have different number of elements")
+
+    def __init__(self):
+        super().__init__()
+
+        box = gui.widgetBox(self.controlArea, box="Pattern")
+        lineedit = gui.lineEdit(box, self, "text")
+        lineedit.returnPressed.connect(self.text_changed)
+        gui.spin(box, self, "skip_lines", 0, 10, label="Skipped lines: ",
+                 callback=self.text_changed)
+        gui.spin(box, self, "use_lines", 1, 10, label="Used lines: ",
+                 callback=self.text_changed)
+        gui.checkBox(box, self, "column_names", label="Column names",
+                     callback=self.text_changed)
+        gui.rubber(self.controlArea)
+
+        gui.auto_commit(self.controlArea, self, "auto_send", "Send",
+                        commit=self.read_file)
+
+        box = gui.hBox(self.mainArea)
+        gui.widgetLabel(box, "Input text")
+        gui.rubber(box)
+        find_line = gui.lineEdit(
+            box, self, "find_text", label="Find: ", orientation=Qt.Horizontal,
+            callback=self.find_changed, callbackOnType=True)
+        find_line.returnPressed.connect(self.find_changed)
+
+        self.in_view = in_view = TextEditWCopy()
+        in_view.setMinimumSize(500, 250)
+        in_view.setLineWrapMode(in_view.NoWrap)
+        in_view.setReadOnly(True)
+        in_view.setFontFamily("Courier")
+        self.mainArea.layout().addWidget(in_view)
+
+        copy = QAction("Copy", in_view, triggered=lambda *x: print("foo") or in_view.copy())
+        copy.setShortcuts(QKeySequence(Qt.Key_C | Qt.ControlModifier))
+        in_view.addAction(copy)
+        for action in in_view.actions():
+            print(action.text())
+
+        gui.widgetLabel(self.mainArea, "Output data")
+        self.out_view = out_view = QTextEdit()
+        out_view.setMinimumSize(500, 250)
+        out_view.setLineWrapMode(out_view.NoWrap)
+        out_view.setReadOnly(True)
+        out_view.setFontFamily("Courier")
+        self.mainArea.layout().addWidget(out_view)
+
+        self.file_changed()
+
+    def file_changed(self):
+        name = "/Users/janez/Desktop/out.out"
```

Comment:
For testing, this should be disabled.

Reply:
`FileNotFoundError` and I cannot open the widget.